operation/manager.go

```diff
@@ -139,21 +151,10 @@ func (sm *SingleOperationManager) WaitForSuccess(
 	}
 }
 
-func (sm *SingleOperationManager) cancelInLock(ctx context.Context) {
```

Comment:
I was able to remove this function in favor of `sm.currentOp != nil -> sm.currentOp.cancelFunc()` because:
* There were only two callers.
* `myOp == op` can no longer be true because:
  * canceling now blocks and
  * both callers already check and early-return if the context has an operation attached to it.

(formal proof left to the reader)